### ISSUE TITLE
communication/azure-communication-sms - cspell found spelling errors in source code

### ISSUE THREAD

**Issue by danieljurek** [Azure/azure-sdk-for-python#22638]
Spell check scanning of package at `sdk/communication/azure-communication-sms` detected spelling errors in the source code and related files. This directory is opted out of PR spell checking in PR #22199 to keep PRs unblocked.

## What to do

1. Ensure Node.js is installed (https://nodejs.org/en/download/).
1. Delete the entry in `.vscode/cspell.json`'s `ignorePaths` field. It will look like: `sdk/communication/azure-communication-sms/**`. You need to do this to enable checking the files.
1. From the root of the repo run spell check using `./eng/common/spelling/Invoke-Cspell.ps1 -ScanGlobs "sdk/communication/azure-communication-sms/**"`
1. Fix detections according to http://aka.ms/azsdk/engsys/spellcheck use the "False positives" section to fix false positives
1. Check in changes (including the change to `.vscode/cspell.json` where the `ignorePaths` is updated to remove the entry for this service.). You may need to run `git add -f .vscode/cspell.json` to force adding the changes to the file in git.

## Spell checking output

```
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/models.py:28:18 - Unknown word (enviornment)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/models.py:32:5 - Unknown word (GCCH)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/models.py:32:13 - Unknown word (GCCH)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/policy.py:11:46 - Unknown word (IOHTTP)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/policy.py:14:33 - Unknown word (IOHTTP)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/user_credential_async.py:53:43 - Unknown word (inprogress)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/user_credential_async.py:77:26 - Unknown word (inprogress)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/user_credential.py:55:37 - Unknown word (inprogress)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/user_credential.py:77:20 - Unknown word (inprogress)
./sdk/communication/azure-communication-sms/azure/communication/sms/_shared/utils.py:27:25 - Unknown word (timegm)
./sdk/communication/azure-communication-sms/tests/test_sms_client_async.py:6:8 - Unknown word (aiounittest)
./sdk/communication/azure-communication-sms/tests/test_sms_client_async.py:24:26 - Unknown word (aiounittest)
```

**Comment by petrsvihlik** [comment]
Note: the files in the `_shared` folder are owned by the ACS Auth team and will be fixed as part of #22636 for all modalities at once

**Comment by petrsvihlik** [comment]
Once fixed, we should remove the package from the spellchecker whitelist - #22199
